nifi-nar-bundles/nifi-py4j-bundle/nifi-py4j-bridge/src/main/java/org/apache/nifi/py4j/StandardPythonBridge.java

```diff
@@ -288,23 +287,18 @@ public String toString() {
         return "StandardPythonBridge";
     }
 
+    private ExtensionId getExtensionId(final String type, final String version) {
+        final List<PythonProcessorDetails> processorTypes = controllerProcess.getController().getProcessorTypes();
+        final Optional<PythonProcessorDetails> processorTypeFound = processorTypes.stream()
+                .filter(details -> details.getProcessorType().equals(type))
+                .filter(details -> details.getProcessorVersion().equals(version))
+                .findFirst();
 
-    private static class ExtensionId {
```

Comment:
Good call changing this to a Record! But I think if it's a record, we don't need the equals or hashCode methods anymore - those are automatically provided.